src/jdk.compiler/share/classes/com/sun/tools/javac/comp/ReflectMethods.java

```diff
@@ -1369,7 +1375,7 @@ public void visitLambda(JCTree.JCLambda tree) {
 
             // Scan the lambda body
             if (tree.getBodyKind() == LambdaExpressionTree.BodyKind.EXPRESSION) {
-                Value exprVal = toValue(tree.body, tree.getDescriptorType(types).getReturnType());
+                Value exprVal = toValue(((JCExpression) tree.body), tree.getDescriptorType(types).getReturnType());
```

Comment:
This cast is probably redundant, as there's no variant of `toValue` that takes `JCStatement` that also takes a second parameter for the target type?

Reply:
Nvm - `tree.body` has type `JCTree`, so you do need a cast :-)